src/business_logic/transaction/internal_objects.rs

```diff
@@ -166,3 +179,376 @@ impl InternalDeploy {
         todo!()
     }
 }
+
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+///  Represents an internal transaction in the StarkNet network that is a declaration of a Cairo
+///  contract class.
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+pub(crate) struct InternalDeclare {
+    pub(crate) class_hash: [u8; 32],
+    pub(crate) sender_address: Address,
+    pub(crate) tx_type: TransactionType,
+    pub(crate) validate_entry_point_selector: Felt,
+    pub(crate) version: u64,
+    pub(crate) max_fee: u64,
+    pub(crate) signature: Vec<Felt>,
+    pub(crate) nonce: Felt,
+    pub(crate) hash_value: Felt,
+}
+
+// ------------------------------------------------------------
+//                        Functions
+// ------------------------------------------------------------
+impl InternalDeclare {
+    pub fn new(
+        contract_class: ContractClass,
+        chain_id: Felt,
+        sender_address: Address,
+        max_fee: u64,
+        version: u64,
+        signature: Vec<Felt>,
+        nonce: Felt,
+    ) -> Result<Self, TransactionError> {
+        let hash = compute_class_hash(&contract_class)?;
+
+        let class_hash = hash
+            .to_bytes_be()
+            .try_into()
+            .map_err(|_| UtilsError::FeltToFixBytesArrayFail(hash.clone()))?;
```

Comment:
This should never fail, otherwise there'd be no need for starknet's keccak variant.

Reply:
Suggestion: use `felt_to_hash()` (defined in `src/utils.rs`.

Reply:
done